File: app/src/main/java/com/dingyi/MyLuaApp/network/builder/HttpClientBuilder.java
package com.dingyi.MyLuaApp.network.builder;

import com.dingyi.MyLuaApp.network.callback.HttpCallBack;
import com.dingyi.MyLuaApp.network.client.HttpClient;

public class HttpClientBuilder {
    private Mode mode;
    private String url;

    public HttpClientBuilder get() {
        mode = Mode.GET;
        return this;
    }

    public HttpClientBuilder url(String url) {
        this.url = url;
        return this;
    }

    public void execute(HttpCallBack callBack) {
        HttpClient.get(url, callBack);
    }

    private enum Mode {
        GET, POST, PUT, DELETE
    }

}

File: app/src/main/java/com/dingyi/MyLuaApp/network/builder/HttpClientBuilder.kt
package com.dingyi.MyLuaApp.network.builder

import com.dingyi.MyLuaApp.network.callback.HttpCallBack
import com.dingyi.MyLuaApp.network.client.get

class HttpClientBuilder {

    private var url: String?=null;
    private var mode="GET"

    fun get():HttpClientBuilder {
        mode="GET"
        return this;
    }

    fun url(url: String):HttpClientBuilder {
        this.url=url
        return this;
    }

    fun enqueue(callback: HttpCallBack) {
        if (mode=="GET") {
            url?.let { get(it,callback) }
        }
    }

}